lang/cs/Org.Apache.REEF.Client/API/IREEFClient.cs

```diff
@@ -43,5 +45,13 @@ public interface IREEFClient
         /// <returns>IDriverHttpEndpoint</returns>
         [Unstable("0.13", "Working in progress for what to return after submit")]
         IDriverHttpEndpoint SubmitAndGetDriverUrl(IJobSubmission jobSubmission);
+
+        /// <summary>
+        /// Returns the application status in running the job
+        /// </summary>
+        /// <param name="appId"></param>
+        /// <returns></returns>
+        [Unstable("0.14", "Working in progress for rest API status returned")]
+        Task<State> GetJobFinalStatus(string appId);
```

Comment:
I exposed AppId as a property in IDriverHttpEndpoint which is the return of SubmitAndGetDriverUrl. But based on your last comments, I removed it. The _appId as a local variable is still available in the implementation of IDriverHttpEndpoint object. It is easy to be exposed as public API. 

Now IDriverHttpEndpoint  exposes AppFinalStatus(), which knows appId and then calls IREEFClient.GetJobFinalStatus(string appId). 

Another way is still expose appId in  IDriverHttpEndpoint, client then can use it to call GetJobFinalStatus(string appId); from IREEFClient directly.